packages/backend/src/matter/devices/media-player-device.ts

```diff
@@ -1,17 +1,51 @@
-import { MatterDevice } from "../matter-device.js";
-import { OnOffPlugInUnitDevice } from "@matter/main/devices";
-import { OnOffServer } from "../behaviors/on-off-server.js";
+import {
+  HomeAssistantEntityState,
+  SpeakerDeviceAttributes,
+} from "@home-assistant-matter-hub/common";
+import { SpeakerDevice } from "@matter/main/devices";
 import { BasicInformationServer } from "../behaviors/basic-information-server.js";
 import { IdentifyServer } from "../behaviors/identify-server.js";
 import { HomeAssistantBehavior } from "../custom-behaviors/home-assistant-behavior.js";
+import { OnOffServer } from "../behaviors/on-off-server.js";
+import { MatterDevice } from "../matter-device.js";
+import {
+  LevelControlConfig,
+  LevelControlServer,
+} from "../behaviors/level-control-server.js";
+import { SwitchDevice } from "./switch-device.js";
+
+const volumeLevelConfig: LevelControlConfig = {
+  getValue: (state: HomeAssistantEntityState<SpeakerDeviceAttributes>) => {
+    if (state.attributes.volume_level != null) {
+      return state.attributes.volume_level * 254;
```

Comment:
Is `volume_level` between 0 and 1 ? Most domains in HA use 0-100 instead of 0-1. 

Consider using min and max value too

Reply:
It is: https://www.home-assistant.io/integrations/media_player/#action-media_playervolume_set

I completely agree about using min/max values, but I encountered some strange behaviors when setting them. It seemed to oscillate between 100 and 254 unpredictably. I’ll investigate this further.

Reply:
There's definitely something wrong when setting the values, even to default (0 and 254). I need to do a bit more troubleshooting to get to the bottom of this.

Reply:
But it does work when not setting them? 😅
Of course we should find out why, but as long as it works, I’d be fine to merge it.

Reply:
It does work, yes.. 😕 

I'm also OK with merging the PR as is, and addressing this particular issue in another one. But TBH, it bothers me I haven't been able to figure out what's happening yet, I will take another look but probably not before the weekend.